### PR TITLE
Feature/create delete ensemble

### PR DESCRIPTION
**Pull request by ViliTrusheva**
This pull request introduces new endpoints for creating and deleting ensembles, along with associated validation and error handling. It also includes updates to the service layer and end-to-end tests to support these new functionalities.

### New Endpoints and Validation:

* [`src/ensembles/ensembles.controller.ts`](diffhunk://#diff-edecd290872ff791a7a035cce4eb3af20f6913ad7bb5ce11269ba444edb39db5R2-R17): Added `POST` and `DELETE` endpoints for creating and deleting ensembles. Added validation using `createEnsembleBodySchema` and `validate` function. [[1]](diffhunk://#diff-edecd290872ff791a7a035cce4eb3af20f6913ad7bb5ce11269ba444edb39db5R2-R17) [[2]](diffhunk://#diff-edecd290872ff791a7a035cce4eb3af20f6913ad7bb5ce11269ba444edb39db5R52-R79)
* [`src/ensembles/lib/validation-schemas.ts`](diffhunk://#diff-8c78f13109fd10ea08bd83a9ed4cf48a302bf2570ad1f702ad8592a9f6c99035R1-R33): Defined `createEnsembleBodySchema` for validating the request body of the `POST` endpoint.
* [`src/ensembles/lib/delete-ensemble.ts`](diffhunk://#diff-56cbab37fbf1b6befe9188d8b34f9b33bc0a11b6e18298bfbb3e983705a9d421R1-R13): Added validation for the `DELETE` endpoint to ensure the provided ID is valid.
* [`src/ensembles/lib/post-ensemble.ts`](diffhunk://#diff-d9a53826c1535f25bcbec4db4ffd9fe263ec4957f58828f9d76f047913839a78R1-R13): Added validation for the `POST` endpoint to ensure the provided ID is valid.

### Service Layer Updates:

* [`src/ensembles/ensembles.service.ts`](diffhunk://#diff-a1e666b874b8c496ec98aff81f2981b26f8c2b69456769819e696c592d9966e7L19-R32): Updated the `findById` method to exclude deactivated ensembles. Added `softDeleteOne` method to mark ensembles as deactivated instead of permanently deleting them. [[1]](diffhunk://#diff-a1e666b874b8c496ec98aff81f2981b26f8c2b69456769819e696c592d9966e7L19-R32) [[2]](diffhunk://#diff-a1e666b874b8c496ec98aff81f2981b26f8c2b69456769819e696c592d9966e7L36-R52)

### End-to-End Tests:

* [`test/ensembles.e2e-spec.ts`](diffhunk://#diff-00ee548c2a02ab0ab1867c7254f322e2d098270308ad4be0b117e24d4ba7f995R7): Added tests for the new `POST` and `DELETE` endpoints, including scenarios for successful creation/deletion and validation errors. [[1]](diffhunk://#diff-00ee548c2a02ab0ab1867c7254f322e2d098270308ad4be0b117e24d4ba7f995R7) [[2]](diffhunk://#diff-00ee548c2a02ab0ab1867c7254f322e2d098270308ad4be0b117e24d4ba7f995R72-R91) [[3]](diffhunk://#diff-00ee548c2a02ab0ab1867c7254f322e2d098270308ad4be0b117e24d4ba7f995R220-R270)

### CODE REVIEW

File: src/ensembles/lib/post-ensemble.ts
Comment (tomassabol) [inline src/ensembles/lib/post-ensemble.ts:1]: this file is not being used anywhere. please remove it

File: src/ensembles/lib/delete-ensemble.ts
Comment (tomassabol) [inline src/ensembles/lib/delete-ensemble.ts:1]: this is the same as function `getId()` in
```bash
/src/ensembles/lib/get-id.ts
```
feel free to just reuse that and remove this. 

💡we could have a separate task, where we create a schema for mongodb ObjectId and just use is wherever needed with the new `validate` function

**Review by tomassabol — CHANGES_REQUESTED** [review]

File: src/ensembles/lib/post-ensemble.ts
Comment (ViliTrusheva) [reply src/ensembles/lib/post-ensemble.ts:1]: girl trust me

**Review by tomassabol — APPROVED** [review]
Looks good. Great job, girl 💅

**Pull request merged**